extensions/sql-database-projects/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -151,36 +153,76 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined,
+					const selectedConnectionStringMethod = await vscode.window.showQuickPick(listOfConnectionStringMethods, {
+						canPickMany: false,
+						title: constants.selectConnectionString,
+						ignoreFocusOut: true
+					});
+					if (!selectedConnectionStringMethod) {
+						// User cancelled
+						return;
 					}
-				) ?? '';
 
-				if (!newConnectionStringValue) {
-					// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
-					continue;
-				}
-
-				try {
-					const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, newConnectionStringValue);
-					if (success) {
-						connectionStringSettingName = newConnectionStringSettingName;
+					let connectionString: string = '';
+					if (selectedConnectionStringMethod === constants.userConnectionString) {
+						// User chooses to enter connection string manually
+						connectionString = await vscode.window.showInputBox(
+							{
+								title: constants.enterConnectionString,
+								ignoreFocusOut: true,
+								value: 'Server=localhost;Initial Catalog={db_name};User ID=sa;Password={your_password};Persist Security Info=False',
+								validateInput: input => input ? undefined : constants.valueMustNotBeEmpty
+							}
+						) ?? '';
+					} else {
+						// Let user choose from existing connections to create connection string from
+						let connectionUri: string = '';
+						const connectionInfo = await vscodeMssqlApi.promptForConnection(true);
+						if (!connectionInfo) {
+							// User cancelled return to selectedConnectionStringMethod prompt
+							continue;
+						}
+						try {
+							// TO DO: https://github.com/microsoft/azuredatastudio/issues/18012
+							connectionUri = await vscodeMssqlApi.connect(connectionInfo);
+						} catch (e) {
+							// give an error if unable to connect to selected connection and return to selectedConnectionStringMethod prompt
+							console.warn(e);
+							continue;
+						}
+						try {
+							connectionString = await vscodeMssqlApi.getConnectionString(connectionUri, false);
+						} catch (e) {
+							// go back to select setting quickpick if user escapes from inputting the value in case they changed their mind
+							console.warn(e);
+							void vscode.window.showErrorMessage(constants.failedToGetConnectionString);
+							continue;
+						}
+					}
+					if (connectionString) {
+						try {
+							const success = await azureFunctionsUtils.setLocalAppSetting(path.dirname(projectUri.fsPath), newConnectionStringSettingName, connectionString);
+							if (success) {
+								// exit both loops and insert binding
+								connectionStringSettingName = newConnectionStringSettingName;
+								break;
+							}
+						} catch (e) {
+							// display error message and show select setting quickpick again
+							void vscode.window.showErrorMessage(constants.selectConnectionError + utils.getErrorMessage(e));
```

Comment:
When combining localized strings you generally shouldn't be appending them like this since different languages may have the words in different order (such as RTL languages).

Instead make the string in the loc constants file take in a parameter that is the error message and pass it as a parameter to the localize function there.